File: api/app/schemas/journal.py
"""订单复盘响应模型 —— camelCase 输出"""

from pydantic import BaseModel, ConfigDict, alias_generators


class CamelModel(BaseModel):
    """camelCase 输出统一；内部字段保持 snake_case"""
    model_config = ConfigDict(
        alias_generator=alias_generators.to_camel,
        populate_by_name=True,
        serialize_by_alias=True,
    )


class OrderOut(CamelModel):
    id: int
    symbol: str               # 标的名
    trade_date: str           # 订单日期 YYYY-MM-DD
    pnl: float                # 带符号盈亏（万元，正盈负亏）
    open_logic: str           # 开仓逻辑
    close_logic: str          # 平仓逻辑
    note: str = ""            # 订单笔记
    image_url: str = ""       # 截图访问地址（/uploads/...）


class JournalSummaryOut(CamelModel):
    total: int                # 订单总笔数
    win_count: int            # 盈利笔数
    loss_count: int           # 亏损笔数
    win_rate: float           # 胜率 %（盈利笔 / 总笔）
    total_pnl: float          # 累计盈亏（万元）


class JournalResponse(CamelModel):
    items: list[OrderOut]
    summary: JournalSummaryOut

File: web/src/types/journal.ts
/** 订单复盘类型定义 */

export interface TradeOrder {
  id: number;
  /** 标的名 */
  symbol: string;
  /** 订单日期 YYYY-MM-DD */
  tradeDate: string;
  /** 带符号盈亏（万元，正盈负亏） */
  pnl: number;
  /** 开仓逻辑 */
  openLogic: string;
  /** 平仓逻辑 */
  closeLogic: string;
  /** 订单笔记 */
  note: string;
  /** 截图访问地址 */
  imageUrl: string;
}

export interface JournalSummary {
  total: number;
  winCount: number;
  lossCount: number;
  /** 胜率 % */
  winRate: number;
  /** 累计盈亏（万元） */
  totalPnl: number;
}

export interface JournalResponse {
  items: TradeOrder[];
  summary: JournalSummary;
}

/** 新建订单提交载荷（multipart 表单字段，不含文件） */
export interface OrderFormValues {
  symbol: string;
  tradeDate: string;
  /** 正数（万元） */
  amount: string;
  /** win | loss */
  pnlType: "win" | "loss";
  openLogic: string;
  closeLogic: string;
  note: string;
}
